spinalcordtoolbox/reports/slice.py

```diff
@@ -268,7 +268,7 @@ def _resample_slicewise(self, image, p_resample, type_img, image_ref=None):
         """
         dict_interp = {'im': 'spline', 'seg': 'linear'}
         # Create nibabel object
-        nii = Nifti1Image(image.data, image.hdr.get_best_affine())
+        nii = Nifti1Image(image.data, affine=image.hdr.get_best_affine(), header=image.hdr)
```

Comment:
@sandrinebedard This change should fix:

- https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4420

Reply:
great! thank you @joshuacwnewton

Reply:
I'll test it out early next week!

Reply:
I tested out this command that failed before:

```python
d/sub-DMAim1Test003/ses-spinalcord21Ch/func/run-1$ sct_qc -i sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean.nii.gz -s sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean_CANAL_seg.nii.gz -p sct_deepseg_sc -qc ./qc

--
Spinal Cord Toolbox (git-jn/4408-avoid-nibabel-int64-crashes-bc363f0ce88db31d2edbe1fe4f73117f16c73895*)

sct_qc -i sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean.nii.gz -s sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean_CANAL_seg.nii.gz -p sct_deepseg_sc -qc ./qc
--


*** Generate Quality Control (QC) html report ***
Resample images to 0.6x0.6 mm
Converting image from type 'int64' to type 'float64' for linear interpolation
Image header specifies datatype 'float64', but array is of type 'int64'. Header metadata will be overwritten to use 'int64'.
QcImage: layout with Axial slice
Compute center of mass at each slice
./qc/ses-spinalcord21Ch/func/run-1/sct_deepseg_sc/2024_04_15_104814.760661/background_img.png
Successfully generated the QC results in ./qc/_json/qc_2024_04_15_104814.760661.json

To see the results in a browser, open the following file:
./qc/index.html
```
And it ran smoothly! Thank you @joshuacwnewton